server/live_payment_processor.go

```diff
@@ -174,28 +172,10 @@ func (p *LivePaymentProcessor) probeOne(ctx context.Context, seg *segment) {
 }
 
 func probeSegment(ctx context.Context, seg *segment) (ffmpeg.MediaFormatInfo, error) {
-	pipeReader, pipeWriter, err := os.Pipe()
-	if err != nil {
-		return ffmpeg.MediaFormatInfo{}, err
-	}
-
-	go func() {
-		defer pipeWriter.Close()
-		io.Copy(pipeWriter, bytes.NewReader(seg.segData))
-	}()
-
-	fname := fmt.Sprintf("pipe:%d", pipeReader.Fd())
-	status, info, err := ffmpeg.GetCodecInfo(fname)
-	if err != nil {
-		return ffmpeg.MediaFormatInfo{}, err
-	}
-	if status != ffmpeg.CodecStatusOk {
-		clog.Info(ctx, "Invalid CodecStatus while probing segment", "status", status)
-		return ffmpeg.MediaFormatInfo{}, fmt.Errorf("invalid CodecStatus while probing segment, status=%d", status)
-	}
-
-	// For WebRTC the probing sometimes returns FPS=90000, which is incorrect and causes issues with payment,
-	// so as a hack let's hardcode FPS to 30
-	info.FPS = 30.0
-	return info, nil
+	// Return a constant value to calculate payments based on time intervals rather than input segment pixel data
```

Comment:
Not sure if this is the kind of thing you meant @j0sh about simplifying the API but yeah I was thinking why do we need this `probeSegment` function now, shouldn't we remove it and modify the calling code?

Reply:
yes - we could simplify everything from the outside API downwards, eg no need for this careful reader handling 

https://github.com/livepeer/go-livepeer/blob/96584b63c5bed60cba13183521f1eae01fa61175/server/ai_http.go#L222-L226